analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -131,6 +131,13 @@ private ExplodedNode ProcessBranch(ExplodedNode node, ControlFlowBranch branch)
         {
             return FromFinally(node.FinallyPoint.CreateNext());     // Redirect from finally back to the original place (or outer finally on the same branch)
         }
+        else if (branch.Semantics == ControlFlowBranchSemantics.Rethrow
```

Comment:
Open question, especially if the "Throw" case proves to be troublesome: Do we need to check the semantics?
Let's say we don't know where to go, and we have conditional branching => we could always do the ConditionEvaluated. So it would be like
```cs
        else
        {
            if (branch.Source.BranchValue is { } branchValue) // Only for conditional branching
            {
                var state = SetBranchingConstraints(branch, node.State, branchValue);
                _ = checks.ConditionEvaluated(new(branchValue.ToSonar(), state, false, node.VisitCount, lva.CapturedVariables));
            }
            return null;    // We don't know where to continue
        }

```

Reply:
That seems reasonable. If we have a condition we should always call `ConditionEvaluated`.

Reply:
The proposed code had just one `return null` for the entire method... changing the current 

`else if`, `return`, `else`, `return`

to nested

`else`, `if`, `return`